### PR TITLE
[Geomagic] ADD an inputForceFeedback data in Geomagic

### PR DESCRIPTION
**Pull request by hugtalbot**
An inputForceFeedback data is added in Geomagic so that a force
can be specified as input of the driver.
This force is used only if no LCP is found.

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: applications/plugins/Geomagic/src/GeomagicDriver.cpp [hunk 1]
```diff
@@ -65,18 +87,18 @@ using namespace sofa::defaulttype;
...
-    std::cerr << hdGetErrorString(error->errorCode) << std::endl;
-    std::cerr << "HHD: "<< error->hHD << std::endl;
-    std::cerr << "Error Code: "<< error->hHD << std::endl;
-    std::cerr << "Internal Error Code: "<< error->internalErrorCode<<std::endl;
-    std::cerr << "Message: " << message << std::endl;
+    msg_error("GeomagicDriver") << hdGetErrorString(error->errorCode);
+    msg_error("GeomagicDriver") << "HHD: "<< error->hHD;
+    msg_error("GeomagicDriver") << "Error Code: "<< error->hHD;
+    msg_error("GeomagicDriver") << "Internal Error Code: "<< error->internalErrorCode;
+    msg_error("GeomagicDriver") << "Message: " << message;
```
Comment (guparan) [inline applications/plugins/Geomagic/src/GeomagicDriver.cpp]: `msg_error()` is enough, GeomagicDriver inherits from Base.

Comment (epernod) [reply applications/plugins/Geomagic/src/GeomagicDriver.cpp]: done for the other use case in the class.
But this one is a static method used on a dedicated thread. So the "GeomagicDriver" should stay.

Comment (damienmarchal) [reply applications/plugins/Geomagic/src/GeomagicDriver.cpp]: Not necessarily, you can also do
msg_error(geomdriver) << 

If you have a GeomagicDriver* geomdriver.  The good is that the message emiiter will contains more info that just the string.

**Comment by epernod** [comment]
Sorry @hugtalbot but your PR is changing the behavior of the haptic device at least in the scene: Geomagic-RigidSkull.scn 

- Rebasing master at commit 25f2a56 : plugin works well -> can feel rigid object with the haptic tool.
- Merging your PR at this master stage: You still have contact but the resistance is lowered -> the 3D model doesn't get into the skull but you can continue to push the haptic device.
- Rebasing master at next commit 8667226 : Nearly no contact between the tool and the skull -> totally broken!

I'll continue the investigation.

**Review by epernod — CHANGES_REQUESTED** [review]
Sorry @hugtalbot but your PR is changing the behavior of the haptic device at least in the scene: Geomagic-RigidSkull.scn 

- Rebasing master at commit 25f2a56 : plugin works well -> can feel rigid object with the haptic tool.
- Merging your PR at this master stage: You still have contact but the resistance is lowered -> the 3D model doesn't get into the skull but you can continue to push the haptic device.

- Rebasing master at next commit 8667226 : Nearly no contact between the tool and the skull -> totally broken!


I'll continue the investigation.

**Comment by epernod** [comment]
@hugtalbot could you review my review in commit cb4eb09
I had to split your VecCoord position device into 2 variables. As in fact the original Data is linked to rigid mechanical Object in the scene and so changing its size and adding new position change the behavior of the rigid objects.
I tested your Demo scene and I can still nicely see the omni moving. Well done.

**Review by epernod — APPROVED** [review]

**Comment by epernod** [comment]
[ci-build][with-scene-tests]

**Comment by guparan** [comment]
@epernod The builds were triggered by latest commit, no need for manual launch :wink:

**Comment by epernod** [comment]
yes but not the tests or I'm missing something?
And @hugtalbot changed 2 scenes.

**Comment by guparan** [comment]
Ah yes, didn't see !

**Comment by epernod** [comment]
good to go I think, finally!

**Pull request merged**